common/rfb/H264LibavDecoderContext.cxx

```diff
@@ -222,9 +222,19 @@ void H264LibavDecoderContext::decode(const uint8_t* h264_in_buffer,
 
   int stride;
   pb->getBuffer(rect, &stride);
-  int dst_linesize = stride * pb->getPF().bpp/8;  // stride is in pixels, linesize is in bytes (stride x4). We need bytes
 
-  sws_scale(sws, frame->data, frame->linesize, 0, frame->height, &swsBuffer, &dst_linesize);
+  int height = frame->height;
 
-  pb->imageRect(rect, swsBuffer, stride);
+  // The server may send a frame with a height greater than that of the rect,
```

Comment:
I understand why this is desirable, but it is unfortunately not something mentioned in the specification for this extension. So it can be argued that it is a violation of the protocol to do so.

So at the minimum, we should make it clear that this is legal. And to do so, I think we need some background research of the current implementations of this extensions:

1. Which servers are already sending H.264 frames larger than the RFB rect
2. How are current and historical clients tolerating this. Evidently TigerVNC doesn't. How are other implementations?

Reply:
Okay, sounds like we might get away with an adjustment. Can you submit a clarification for rfbproto?

In that spirit, perhaps best specify that a smaller frame is invalid? Otherwise, it would need to be specific what to do with the missing pixels.

Reply:
Done. I agree that smaller frames should be invalid.